tests/test_diagnostic_response.py

```diff
@@ -0,0 +1,36 @@
+from lennoxs30api.s30api_async import lennox_zone, s30api_async, lennox_system
+from lennoxs30api.lennox_home import lennox_home
+
+import json
+import os
+
+
+# where device responses can be found
+script_dir = os.path.dirname(__file__) + "/messages/"
+
+
+def setup_load_configuration() -> s30api_async:
+    api = s30api_async("myemail@email.com", "mypassword", None, ip_address="10.0.0.1")
+    api.setup_local_homes()
+
+    file_path = os.path.join(script_dir, "device_response_lcc.json")
+    with open(file_path) as f:
+        data = json.load(f)
+    api.processMessage(data)
+
+    return api
+
+
+def test_process_equipment_energy():
+    api = setup_load_configuration()
+    lsystem: lennox_system = api.getSystems()[0]
+    assert lsystem.diagInverterInputVoltage == None
+    assert lsystem.diagInverterInputCurrent == None
+
+    file_path = os.path.join(script_dir, "equipments_response_energy.json")
+    with open(file_path) as f:
+        data = json.load(f)
+    api.processMessage(data)
+
+    assert lsystem.diagInverterInputVoltage == 247.0
+    assert lsystem.diagInverterInputCurrent == 4.301
```

Comment:
+10 on adding a unit test.  We can also test the dirty processing in here.  Take a look at the changes made in the mutation test for an example.

Reply:
Added some tests for failure modes and to verify some update callback behavior